### ISSUE TITLE
URL equality doubts

### ISSUE THREAD

**Issue by coutoPL** [lemonlabsuk/scala-uri#282]
**Description**
An order of query params are taken into consideration during the URI instances comparison. It looks it's a result of choosing Vector as a structure for storing query params. The same problem applies to the cats' Eq instance (` implicit val eqUri: Eq[Uri] = Eq.fromUniversalEquals`)

I'm not sure if this was done intentionally or not. Seems that the specification doesn't say much about it. Most of java/scala libs I tested, work the same, but to me, the order of query parameters should not matter. What are your thoughts about this problem? At least it'd be nice to define alternative `Eq[Uri]` instance which supports the comparison I mentioned about.

**Example**
```scala
@ import io.lemonlabs.uri.Url
import io.lemonlabs.uri.Url

@ Url.parse("http://example.com?one=1&two=2")
res3: Url = AbsoluteUrl("http", Authority(None, DomainName("example.com"), None), , QueryString(Vector(("one", Some("1")), ("two", Some("2")))), None)

@ Url.parse("http://example.com?two=2&one=1")
res4: Url = AbsoluteUrl("http", Authority(None, DomainName("example.com"), None), , QueryString(Vector(("two", Some("2")), ("one", Some("1")))), None)

@ res3 == res4
res5: Boolean = false
```

**Comment by theon** [comment]
Hi @coutoPL, thanks for raising this 👍  I agree that there are situations where the order of the query params won't matter for equality.

The Vector of tuples was deliberately chosen (it changed from a Map to Vector in scala-uri 0.3 -> 0.4) so that `Url.parse(str).toString` maintained the order of query parameters for users who don't want them rearranged. 

I think your suggestion of adding a second `Eq[Uri]` instance is the best option and am happy to add this. I'll also add a method something like `url.equalsQueryUnordered(otherUrl)` which will do the same thing for users who are not using Cats.

**Comment by coutoPL** [comment]
Great! 
Thanks @theon for the quick response

**Comment by coutoPL** [comment]
And one more thing:
I was wondering that if the standard doesn't specify it, maybe it would be nice to explicitly say (in docs) what is a chosen strategy? None of libs, which I know, do that, but IMO it's worth to mention, because it's not clear by default (it really depends on implementation decisions).

**Comment by theon** [comment]
Yes, makes sense. We can also make sure it is clear in some scaladoc comments

**Comment by theon** [comment]
I've released a first version of this in `3.2.0`. It is documented here: https://github.com/lemonlabsuk/scala-uri#url-equality

Hope it fits your needs. Let me know how you get on 👍

**Comment by coutoPL** [comment]
Thanks a lot. As a workaround I created unordered Eq in my codebase. Will migrate to scala-uri unordered Eq when I upgrade the lib in the future.
